x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.test.ts

```diff
@@ -206,27 +206,42 @@ describe('rule_converters', () => {
   });
 
   describe('convertPatchAPIToInternalSchema', () => {
-    test('should not update version for immutable rules', () => {
-      const patchParams = {
+    test('should set version to one specified in next params', () => {
+      const nextParams = {
+        index: ['new-test-index'],
+        language: 'lucene',
+        version: 3,
+      };
+      const existingRule = getRuleMock({ ...getQueryRuleParams() });
+      const patchedParams = convertPatchAPIToInternalSchema(nextParams, existingRule);
+      expect(patchedParams).toEqual(
+        expect.objectContaining({
+          params: expect.objectContaining({ version: 3 }),
+        })
+      );
+    });
+
+    test('should not update version for immutable rules if it is not specified in next params', () => {
```

Comment:
Do you think it would make sense to add one more test case, complimentary to this that goes like `SHOULD update version for immutable rules if specified in params` ?

From the new logic:
```js
      version: nextParams.version
        ? nextParams.version
        : shouldUpdateVersion(nextParams, existingParams)
        ? existingParams.version + 1
        : existingParams.version,
```
my understanding is that the version will be updated if it is defined in `nextParams` even for the cases in which the rules are immutable.

Reply:
Yea, makes sense, added a separate test case for that